docs: Polish the langref concepts (#2241)
slint-ui/slint: docs/langref/src/concepts/file.md

```

Both `MyButton` and `MyApp` are components. `Window` and `Rectangle` are built-in elements
used by `MyApp`. `MyApp` also re-uses the `MyButton` component as two separate elements.

> Elements have properties, which you can assign values to. Here the first `MyButton`
> element has a property `text` that we assign a string constant "hello" to. You
> can assign expressions using other properties as well. Slint will re-evaluate
> properties when any of the properties they depend on change, which makes the
> user-interface reactive.

Reviewer: `MyBytton` has a `text` property only through its inheritance, so you could argue that `Text` is the one that has the property. I'd omit the "ownership" part here, to avoid confusion, and go straight to the assignment. We've already established that every element has properties after all :)

```suggestion
Elements have properties, which you can assign values to. Here we assign a string
constant "hello" to the first `MyButton`'s `text` property. You
can also assign entire expressions. Slint will re-evaluate the expressions when any
of the properties they depend on change, which makes the user-interface reactive.
```

}
```

Names have to be valid [identifiers](../identifiers.md).

Some elements are also accessible under pre-defined names:

-   `root` refers to the outermost element of a component.
-   `self` refers to the current element.
-   `parent` refers to the parent element of the current element.

> Users can not redefine these names to refer to something else.

Reviewer: How about something even simpler (and using a contraction). This way the emphasis is on the fact that the names are reserved, and the consequence follows after that (not being possible to re-defined), instead of omitting the reason why the user can't re-define them (they are reserved).

```suggestion
These names are reserved and you can't be re-define them.
```
